test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs

```diff
@@ -892,5 +893,129 @@ public void InstallCommand_DependencyResolutionFailure()
                 Util.DeleteDirectory(source);
             }
         }
+
+        // Tests that when credential is saved in the config file, it will be passed 
+        // correctly to both the index.json endpoint and registration endpoint, even
+        // though one uri does not start with the other uri.
+        [Fact]
+        public void InstallCommand_AuthenticatedV3WithCredentialSavedInConfig()
+        {
+            var nugetexe = Util.GetNuGetExePath();
+            var randomTestFolder = TestFilesystemUtility.CreateRandomTestFolder();
+            bool credentialsPassedToRegistrationEndPoint = false;
+
+            try
+            {
+                // Server setup                
+                using (var serverV3 = new MockServer())
+                {
+                    var registrationEndPoint = serverV3.Uri + "w";
+                    var indexJson = JObject.Parse(@"{
+                  ""version"": ""3.2.0"",
+                  ""resources"": [ {""@id"": """ + registrationEndPoint + @""", ""@type"": ""RegistrationsBaseUrl/3.0.0-beta""} ],
+                ""@context"": {
+                ""@vocab"": ""http://schema.nuget.org/services#"",
+                ""comment"": ""http://www.w3.org/2000/01/rdf-schema#comment""
+                    }}");
+
+                    serverV3.Get.Add("/a/b/c/index.json", r =>
+                    {
+                        var h = r.Headers["Authorization"];
+                        var credential = String.IsNullOrEmpty(h) ? 
+                            null :
+                            System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
+
+                        if (StringComparer.OrdinalIgnoreCase.Equals("test_user:test_password", credential))
+                        {
+                            return new Action<HttpListenerResponse>(response =>
+                            {
+                                response.StatusCode = (int)HttpStatusCode.OK;
+                                response.ContentType = "text/javascript";
```

Comment:
This was copied from another test. In fact the content type is never checked by our client.